src/sparse_variational.jl

```diff
@@ -104,6 +160,96 @@ function StatsBase.mean_and_var(
     return μ, Σ_diag
 end
 
+#
+# NonCentred parametrisation.
+#
+
+raw"""
+    posterior(sva::SparseVariationalApproximation{NonCentred})
+
+Compute the approximate posterior [1] over the process `f =
+sva.fz.f`, given inducing inputs `z = sva.fz.x` and a variational
+distribution over inducing points `sva.q` (which represents ``q(ε)``
+where `ε = cholesky(cov(fz)).U' \ (f(z) - mean(f(z)))`). The approximate posterior at test
+points ``x^*`` where ``f^* = f(x^*)`` is then given by:
+
+```math
+q(f^*) = \int p(f | ε) q(ε) du
+```
+which can be found in closed form.
+
+[1] - Hensman, James, Alexander Matthews, and Zoubin Ghahramani. "Scalable
+variational Gaussian process classification." Artificial Intelligence and
+Statistics. PMLR, 2015.
+"""
+function AbstractGPs.posterior(approx::SparseVariationalApproximation{NonCentred})
+    fz = approx.fz
+    data = (Cuu=_chol_cov(fz), C_ε=_chol_cov(approx.q))
+    return ApproxPosteriorGP(approx, fz.f, data)
+end
+
+#
+# Various methods implementing the Internal AbstractGPs API.
+# See AbstractGPs.jl API docs for more info.
+#
+
+# Produces a matrix that is consistently referred to as A in this file. A more descriptive
+# name is, unfortunately, not obvious. It's just an intermediate quantity that happens to
+# get used a lot.
+_A(f, x) = f.data.Cuu.U' \ cov(f.prior, inducing_points(f), x)
+
+function Statistics.mean(
+    f::ApproxPosteriorGP{<:SparseVariationalApproximation{NonCentred}}, x::AbstractVector
+)
+    return mean(f.prior, x) + _A(f, x)' * mean(f.approx.q)
```

Comment:
why not do as for `Centred` case and include the solve on Cuu.L within the \alpha ? then maybe these functions could even be the same for both Centred and NonCentred?

Reply:
Fair point -- it's probably a reasonable time / memory tradeoff. I've always felt that it doesn't buy you a lot since you usually want `mean_and_var` if you're using a GP, but I suppose that if you really do just want the mean it's reasonable to want it.

Reply:
well I commented here but doesn't it similarly apply to var/cov and mean_and_var/cov as well?

Reply:
Nah. In those functions you need to invoke `_A` because you need it in order to compute the `var` / `cov` efficiently.

Reply:
The relevant part of the equations are:
- centred: `Kfu * (inv(Kuu) - inv(Kuu) * S * inv(Kuu)) * Kuf`
- non-centred: `Kfu * (inv(Kuu) - inv(cholesky(Kuu).L)' * S * inv(cholesky(Kuu).L)) * Kuf`
either way, we _could_ phrase this as `Kfu * A * Kuf` where `A` is precomputed... in GPflow we had a long discussion about doing this (it's faster for predictions) vs. not doing this (it might not be faster within the training loop, and it might not be as numerically stable)...

Reply:
Ah I see -- I hadn't thought to chop it up that way, I guess because I assumed the numerics wouldn't be great (as you suggest).

Unless you feel strongly, I'd like to address the precise way in which we opt to compute this in a separate issue -- any chance will be non-breaking, so it won't cause problems changing it if we obtain evidence later on that ordering the computations differently would be better.

Reply:
I'm going to resolve this -- we can fix up in future if needed.